### ISSUE TITLE
Port EmailLogDao and EmailLogEntry to Eloquent Model

### ISSUE THREAD

**Issue by taslangraham** [pkp/pkp-lib#10129]
**Describe the change**
Right now the EmailLogEntry entity is a DataObject and it follows the DAO pattern.

We will port EmailLogEntry entity and its respective DAO class, and related sub-classes to Laravel's Eloquent Model..

**What application will be affected?**
OJS, OMP or OPS main branch

### PRs:
**pkp-lib main** -  https://github.com/pkp/pkp-lib/pull/10150/files
- Introduced [eloquence](https://github.com/kirkbushell/eloquence) package to allow easy camelCase access to Laravel eloquent model attributes
- Converted EmailLogEntry class to Laravel eloquent model
- Replaced DAOs with a repository
- Replaced child classes of `EmailLogEntry` with enums that contain app-specific values

**OMP main** - https://github.com/pkp/omp/pull/1632/files
**OJS main** - https://github.com/pkp/ojs/pull/4369/files
**OPS main** - https://github.com/pkp/ops/pull/732/files

**Comment by taslangraham** [comment]
Ready for review @Vitaliy-1 @asmecher 
pkp-lib main - https://github.com/pkp/pkp-lib/pull/10150/files
OMP main - https://github.com/pkp/omp/pull/1632/files

**Comment by asmecher** [comment]
Thanks, @taslangraham, I've added [a review here](https://github.com/pkp/pkp-lib/pull/10150#pullrequestreview-2175998033)!

**Comment by asmecher** [comment]
Added a comment over on https://github.com/pkp/pkp-lib/pull/10150#issuecomment-2232030186, thanks, @taslangraham!

**Comment by taslangraham** [comment]
All updated @asmecher 
- https://github.com/pkp/omp/pull/1632/commits/914c87ce0cbae3f478de0f38439f340b2c743a10#diff-d6dbec6efe4425466bdbd9bd101a861b698709f6fd33f1a3a2d1c3d86dde3c60
- https://github.com/pkp/pkp-lib/pull/10150/commits/e6f110bcd5151b3f9fdc67cbbe86d7f9df667336#diff-0b46927ec2829a68ba70e80024a7073623a16c2ea4ddea6abf7ad5bb0f8113a4

**Comment by asmecher** [comment]
Merged all -- thanks, @taslangraham! I made two additional changes, not wanting to hold up the merge:
- [Removed `MonographFileEmailLogEventType.php` entirely](https://github.com/pkp/omp/commit/ee4155a55c0997b5a977b7311822f67ab3778965) (OMP) -- it's dead code.
- [Renamed the `SUBMISSION_EMAIL_` prefix from members of `SubmissionEmailLogEventType`](https://github.com/pkp/pkp-lib/commit/34574ebd86ccc1055833f0aae449b95d55621525) -- now that these are in a class and backwards compatibility is broken, it's a good time to make them a little more concise.

Thanks again -- this is a good clean-up of some very old code!
